backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java

```diff
@@ -475,13 +485,17 @@ public TestEvent createMultiplexTestEvent(
     resultRepository.save(fluB);
     order = testOrderRepository.save(order);
 
-    TestEvent event = new TestEvent(order, hasPriorTests, Set.of(covid, fluA, fluB));
+    TestEvent event = new TestEvent(order, hasPriorTests);
     testEventRepository.save(event);
-    covid.setTestEvent(event);
+
+    Set<Result> copiedResults =
+        List.of(covid, fluA, fluB).stream().map(Result::new).collect(Collectors.toSet());
+    copiedResults.forEach(result -> result.setTestEvent(event));
+    resultRepository.saveAll(copiedResults);
+    event.getResults().addAll(copiedResults);
+
     resultRepository.save(covid);
-    fluA.setTestEvent(event);
     resultRepository.save(fluA);
-    fluB.setTestEvent(event);
     resultRepository.save(fluB);
```

Comment:
we already saved these at the very beginning of this method right?

Reply:
oops forgot to remove them